### PR TITLE
new storage backend

### LINKED ISSUE #5: use player meta instead of global save file
needs a compatibility layer, because player meta can only be edit for online players. (maybe mod storage?)

### PR DESCRIPTION
**Pull request by Niklp09**
Closes https://github.com/mt-mods/dreambuilder_hotbar/issues/5

Uses wsor's suggestion:
file -> modstorage -> player meta

### CODE REVIEW

File: init.lua [hunk 1]
```diff
@@ -1,60 +1,89 @@
...
 end
 
-local player_hotbar_settings = {}
-local f = io.open(minetest.get_worldpath()..DIR_DELIM.."hotbar_settings","r")
-if f then
-	player_hotbar_settings = minetest.deserialize(f:read("*all"))
-	f:close()
+local base_img = themename.."gui_hb_bg_1.png"
+local imgref_len = string.len(base_img) + 8 -- accounts for the stuff in the string.format() below.
+local hotbar_size_default = 16
```
Comment (SwissalpS) [inline init.lua:10]: if this refers to the number of slots shown by default, it should be 8 not 16

Comment (wsor4035) [reply init.lua:10]: the above comment is invalid

**Comment by wsor4035** [comment]
briefly looked at this. unless i missed something, it seems the migration runs ever time the world is started? doesnt exactly seem ideal....

File: init.lua (hunk 1 above)
Comment (SwissalpS) [reply init.lua:10]: why? I was referring to line 10: hotbar_size_default

Comment (wsor4035) [reply init.lua:10]: it has always been 16 for this mod. thus your comment is invalid

File: init.lua [hunk 2]
```diff
@@ -1,60 +1,89 @@
...
+	local size = meta:get_int("hotbar_size")
+	if size == 0 then -- not present
+		size = storage:get_int(name)
+		if size ~= 0 then -- migrate
+			storage:set_string(name, "")
+			meta:set_int("hotbar_size", size)
+		end
+	end
+	if size == 0 then size = nil end
+	return size or hotbar_size_default
```
Comment (OgelGames) [inline init.lua]: Could be simplified to this: (fine either way though)
```suggestion
	return size > 0 and size or hotbar_size_default
```

**Review by OgelGames — APPROVED** [review]

**Comment by OgelGames** [comment]
> briefly looked at this. unless i missed something, it seems the migration runs ever time the world is started? doesnt exactly seem ideal....

Only if the file exists. (the file is removed after migration)

**Pull request merged**